src/TwoOutOfThreeColumnVisitor.cpp

```diff
@@ -0,0 +1,148 @@
+#include "TwoOutOfThreeColumnVisitor.hpp"
+#include "Grid.hpp"
+#include "RegionHolder.hpp"
+#include <vector>
+#include <iostream>
+using namespace std;
+
+TwoOutOfThreeColumnVisitor::TwoOutOfThreeColumnVisitor(){}
+
+
+// For local use only
+pair<int,int> findOnlyOneNegativeInCols(std::vector<std::vector<bool>> possibles);
+
+bool TwoOutOfThreeColumnVisitor::Visit(Grid &ioGrid) const{
+        bool changed = false;
+        std::vector<RegionHolder> regions;
+        std::vector<std::vector<bool>> possibles;
+        std::vector<bool> possibleColumns;
+        pair<int,int> position;
+        int col;
+        std::vector<bool>::iterator it;
+        TripleHolder fillableTriplet;
+
+        for(int main_col=0; main_col<3;++main_col) {
+                // vertical
+                
+                for(int i=0; i<3; ++i) {
+                        // work with the ith region in this row
+                        regions = std::vector<RegionHolder>();
+                        regions.push_back(ioGrid.getRegion(3*((i+0)%3) + main_col));
+                        regions.push_back(ioGrid.getRegion(3*((i+1)%3) + main_col));
+                        regions.push_back(ioGrid.getRegion(3*((i+2)%3) + main_col));
+                        
+                        for(int value = 1; value<=9; ++value) {
+                                // test for all different values
+                                //
+                                if(regions[0].isValuePresent(value)){continue;}
+
+                                // value is not present in the ith region
+                                // find the col where it should be
+                                
+                                possibleColumns = vector<bool>(3, false);
+                                position = regions[1].valuePosition(value);
+                                if (position.second != -1) {
+                                        possibleColumns[position.second] = true;
+                                }
+                                position = regions[2].valuePosition(value);
+                                if (position.second != -1) {
+                                        possibleColumns[position.second] = true;
+                                }
```

Comment:
By default it is false, and you store in the "possibleColumns" true when it is impossible? Maybe the opposite?